library/src/com/google/maps/android/kml/KmlRenderer.java

```diff
@@ -0,0 +1,870 @@
+     * @param placemarkVisibility Boolean value, where true indicates the placemark geometry is
+     *                            shown initially on the map, false for not shown initially on the
+     *                            map.
+     * @return Google Map Object of the placemark geometry after it has been added to the map.
+     */
+    private Object addPlacemarkToMap(KmlPlacemark placemark, Boolean placemarkVisibility) {
+        String placemarkId = placemark.getStyleID();
+        KmlGeometry geometry = placemark.getGeometry();
+        KmlStyle style = getPlacemarkStyle(placemarkId);
+        KmlStyle inlineStyle = placemark.getInlineStyle();
+        return addToMap(placemark, geometry, style, inlineStyle, placemarkVisibility);
+    }
+
+    /**
+     * Adds placemarks with their corresponding styles onto the map
+     *
+     * @param kmlContainers An arraylist of folders
+     */
+    private void addContainerGroupToMap(Iterable<KmlContainer> kmlContainers,
+            boolean containerVisibility) {
+        for (KmlContainer container : kmlContainers) {
+            Boolean isContainerVisible = getContainerVisibility(container, containerVisibility);
+            if (container.getStyles() != null) {
+                // Stores all found styles from the container
+                mStylesRenderer.putAll(container.getStyles());
+            }
+            if (container.getStyleMap() != null) {
+                // Stores all found style maps from the container
+                assignStyleMap(container.getStyleMap(), mStylesRenderer);
+            }
+            addContainerObjectToMap(container, isContainerVisible);
+            if (container.hasNestedKmlContainers()) {
+                addContainerGroupToMap(container.getNestedKmlContainers(), isContainerVisible);
+            }
+        }
+    }
+
+    /**
+     * Goes through the every placemark, style and properties object within a <Folder> tag
+     *
+     * @param kmlContainer Folder to obtain placemark and styles from
+     */
+    private void addContainerObjectToMap(KmlContainer kmlContainer, boolean isContainerVisible) {
+        Set<KmlPlacemark> containerPlacemarks = kmlContainer.getPlacemarks().keySet();
+        for (KmlPlacemark placemark : containerPlacemarks) {
+            Boolean isPlacemarkVisible = getPlacemarkVisibility(placemark);
+            Boolean isObjectVisible = isContainerVisible && isPlacemarkVisible;
+            Object mapObject = addPlacemarkToMap(placemark, isObjectVisible);
+            kmlContainer.setPlacemark(placemark, mapObject);
+        }
+    }
+
+    /**
+     * Obtains the styleUrl from a placemark and finds the corresponding style in a list
+     *
+     * @param styleId StyleUrl from a placemark
+     * @return Style which corresponds to an ID
+     */
+    private KmlStyle getPlacemarkStyle(String styleId) {
+        KmlStyle style = mStylesRenderer.get(null);
```

Comment:
with the code as it is, you can just have:

return mStylesRenderer.get(styleId);